simulation/world.py

```diff
@@ -263,9 +263,13 @@ def tick(self) -> bool:
                     visible_agents.append(simulation.vision.AgentView(other_agent))
             
             # and run the agent code
-            agent.tick(seen_agents=visible_agents, noises=[])
+            agent.tick(seen_agents=visible_agents)
         self._collision_check()
 
+        for ID, agent in self.agents.items():
+            agent.on_noise(self.noises);
```

Comment:
I know it's tempting to do this, but it makes the code a lot harder to read, try to pass all the info to the `agent.tick` method and do the processing and calling of event handlers there if possible.

Reply:
If i recall corretly, doing it that way caused agents not to hear noise of other agents.
The first agent only precieved his noise, the second the one of the first and its own etc.

Reply:
if you just pass all noises in the `world` to the `tick` method and then filter down which ones the agent should be able to hear in there then it should be fine I think